source/grid/manifold_lib.cc

```diff
@@ -2187,17 +2207,24 @@ TransfiniteInterpolationManifold<dim, spacedim>::
             cell->real_to_unit_cell_affine_approximation(points[i]);
           current_distance += GeometryInfo<dim>::distance_to_unit_cell(point);
         }
-      distances_and_cells.emplace_back(current_distance, cell->index());
```

Comment:
The error here is
```
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/vector.hpp:2745:1: error: use of deleted function ‘boost::container::container_detail::insert_emplace_proxy_arg2<boost::container::                       small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>::insert_emplace_proxy_arg2(const boost::container::container_detail::                 insert_emplace_proxy_arg2<boost::container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>&)’
        this->priv_forward_range_insert_new_allocation
  ^
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/detail/advanced_insert_int.hpp:399:7161: note: ‘boost::container::container_detail::insert_emplace_proxy_arg2<boost::container::                          small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>::insert_emplace_proxy_arg2(const boost::container::container_detail::                 insert_emplace_proxy_arg2<boost::container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>&)’ is implicitly deleted because the      default definition would be ill-formed:
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/detail/advanced_insert_int.hpp:399:7161: error: use of deleted function ‘boost::container::container_detail::insert_nonmovable_emplace_proxy2<boost::     container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>::insert_nonmovable_emplace_proxy2(const boost::container::                 container_detail::insert_nonmovable_emplace_proxy2<boost::container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>&)’
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/detail/advanced_insert_int.hpp:399:6004: note: ‘boost::container::container_detail::insert_nonmovable_emplace_proxy2<boost::container::                   small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>::insert_nonmovable_emplace_proxy2(const boost::container::container_detail::          insert_nonmovable_emplace_proxy2<boost::container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>&)’ is implicitly deleted because   the default definition would be ill-formed:
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/detail/advanced_insert_int.hpp:399:6004: error: copying non-static data member ‘int&& boost::container::container_detail::                                insert_nonmovable_emplace_proxy2<boost::container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>::m_p1’ of rvalue reference type
 /export/home/darndt/dealii/bundled/boost-1.62.0/include/boost/container/vector.hpp:2986:1: note:   initializing argument 5 of ‘void boost::container::vector<T, Allocator>::priv_forward_range_insert_new_allocation(T*, boost::  container::vector<T, Allocator>::size_type, T*, typename boost::container::allocator_traits<Allocator>::size_type, InsertionProxy) [with InsertionProxy = boost::container::container_detail::insert_emplace_proxy_arg2<boost::   container::small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >, std::pair<double, unsigned int>*, double&, int>; T = std::pair<double, unsigned int>; Allocator = boost::container::       small_vector_allocator<boost::container::new_allocator<std::pair<double, unsigned int> > >; boost::container::vector<T, Allocator>::size_type = long unsigned int; typename boost::container::allocator_traits<Allocator>::       size_type = long unsigned int]’
     void priv_forward_range_insert_new_allocation
  ^
```

Reply:
Does `push_back` also not work?

Reply:
Yes, seems so. Just too simple for me. :wink:

Reply:
Admittedly, passing all the code through the `CUDA` frontend is pretty unusual. I would say it's more `CUDA` to blame.